### PR TITLE
feat(U1): add TypeId<> and TypeIdVector<>

### PR DESCRIPTION
**Pull request by ledouxm**
This is a copy of [this closed PR](https://github.com/bot4dofus/Datafus/pull/36) since i had some issues with the fork...

This is the original description for context: 

Hi, i also developped a tool that does pretty much the same but i think having a single repo which we rely on could be a great thing for the ecosystem.

Sadly, as this [issue ](https://github.com/bot4dofus/Datafus/issues/29) states, the events.json file isn't complete enough to read a message.

This PR aims to fix this by adding two new types along with Vector : 
- "TypeId<BaseEntity>" means the entity we have to read can be identified by its id that we get reading a short from the buffer
- "TypeIdVector<Short, BaseEntity>" means the buffer contains a vector with a length encoded in a short, and for every item of the array we have to read a short to get the id of the matching entity.

I can clarify any point if needed !

### CODE REVIEW

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -175,7 +175,7 @@ jobs:
 
       - name: Get latest Dofus version
         run: |
-          export DOFUS_VERSION=$(cat data/U/version/VERSION)
```
Comment (LucBerge) [inline .github/workflows/workflow.yml:178]: I forgot this on the main branch

**Comment by LucBerge** [comment]
No need to add the .json files in the PR

**Comment by ledouxm** [comment]
do you want the en and fr ones  or i delete them too ?

**Comment by LucBerge** [comment]
Nothing in `data/` must be manually commited. Remove *.json please

**Comment by LucBerge** [comment]
> So everytime you push something you have to delete the json files populated by the github action ?

No, its a mistake I made... By delete, I mean checkout the *.json to the previous commit so that it does not appear in the PR. I tried to do it on the UI but with no success...

**Comment by ledouxm** [comment]
ok i've rollbacked them, it feels weird having json files not matching the ones generated by the code tho

**Comment by ledouxm** [comment]
do you plan on merging this PR ? 
if it doesnt fit your needs, tell me what to change

File: src/A2.py [hunk 2]
```diff
@@ -137,7 +151,7 @@ def parse(self):
                                 continue
 
         return {
-            'file': self.file_name,
+            'file': self.file_name.replace("\\", "/"),
```
Comment (LucBerge) [inline src/A2.py:154]: Remove. The script is automatically run on unix machines. I guess you used it on windows to check diffs?

File: .github/workflows/workflow.yml [hunk 3]
```diff
@@ -175,7 +175,7 @@ jobs:
...
+          export DOFUS_VERSION=$(cat data/U/VERSION)
```
Comment (LucBerge) [inline .github/workflows/workflow.yml:178]: Remove

File: src/A2.py [hunk 4]
```diff
@@ -61,10 +70,14 @@ def addType(self, _type):
...
+    def hasTypeId(self):
+        return self._has_type_id
+
     def buildSocketType(self):
-        self._socket_type = str(self._types[0])
+        self._socket_type =  "TypeId<" + str(self._types[0]) + ">" if self.hasTypeId() and len(self._types) < 2 else str(self._types[0]) 
+        
         for i in range(1, len(self._types)):
-            self._socket_type = "Vector<" + self._socket_type + "," + str(self._types[i]) + ">"
+            self._socket_type = ("TypeIdVector<" if self.hasTypeId() else "Vector<") + self._socket_type + "," + str(self._types[i]) + ">"
```
Comment (LucBerge) [inline src/A2.py:80]: Call `self._has_type_id` instead of `self.hasTypeId()`

File: src/A2.py [hunk 5]
```diff
@@ -61,10 +70,14 @@ def addType(self, _type):
         except ValueError:
             return False
 
+    def hasTypeId(self):
```
Comment (LucBerge) [inline src/A2.py:73]: Remove the method

**Comment by LucBerge** [comment]
@ledouxm A tricky thing append while removing a branch: This issue has been automatically closed

**Pull request closed without merging**